java/src/main/java/io/opentracing/grpc/ClientTracingInterceptor.java

```diff
@@ -0,0 +1,290 @@
+        this.tracer = tracer;
+        this.operationName = "";
+        this.streaming = false;
+        this.verbose = false;
+        this.tracedAttributes = new HashSet<ClientRequestAttribute>();
+    }
+
+    private ClientTracingInterceptor(Tracer tracer, String operationName, boolean streaming,
+        boolean verbose, Set<ClientRequestAttribute> tracedAttributes) {
+        this.tracer = tracer;
+        this.operationName = operationName;
+        this.streaming = streaming;
+        this.verbose = verbose;
+        this.tracedAttributes = tracedAttributes;
+    }
+
+    /**
+     * Use this interceptor to trace all requests made by this client channel.
+     * @param channel to be traced
+     */ 
+    public Channel intercept(Channel channel) {
+        return ClientInterceptors.intercept(channel, this);
+    }
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+        MethodDescriptor<ReqT, RespT> method, 
+        CallOptions callOptions, 
+        Channel next
+    ) {
+        final String operationName;
+        if (this.operationName.equals("")) {
+            operationName = method.getFullMethodName();
+        } else {
+            operationName = this.operationName;
+        }
+
+        Span activeSpan = OpenTracingContextKey.activeSpan();
+        final Span span = createSpanFromParent(activeSpan, operationName);
+        System.out.println("Started span " + span.toString());
+
+        for (ClientRequestAttribute attr : this.tracedAttributes) {
+            switch (attr) {
+                case AFFINITY:
+                    if (callOptions.getAffinity() == null) {
+                        span.setTag("Affinity", "null");
+                    } else {
+                        span.setTag("Affinity", callOptions.getAffinity().toString());
+                    }  
+                    break;
+                case ALL_CALL_OPTIONS:
+                    span.setTag("Call Options", callOptions.toString());
+                    break;
+                case AUTHORITY:
+                    if (callOptions.getAuthority() == null) {
+                        span.setTag("Authority", "null");
+                    } else {
+                        span.setTag("Authority", callOptions.getAuthority());                        
+                    }
+                    break;
+                case COMPRESSOR:
+                    if (callOptions.getCompressor() == null) {
+                        span.setTag("Compressor", "null");
+                    } else {
+                        span.setTag("Compressor", callOptions.getCompressor());
+                    }
+                    break;
+                case DEADLINE:
+                    if (callOptions.getDeadline() == null) {
+                        span.setTag("Deadline", "null");
+                    } else {
+                        span.setTag("Deadline", callOptions.getDeadline().toString());
```

Comment:
let's do `gprc.deadline_millis` and have the value be an int rather than a string